### DISCUSSION TITLE
Filter realtime messages by foreign table column

### DISCUSSION THREAD

**Discussion by jojomatik** [supabase/supabase#27185] [Questions] [answered]
I want to filter realtime updates from tables `order` and `order_article` by a certain customer.

For table `order` I use the following query:
```ts
    .on(
      "postgres_changes",
      {
        event: "INSERT",
        schema: "public",
        table: "order",
        filter: "customer_id=eq." + customerId,
      },
```

For table `order_article` I only want to receive updates for article states for that customer. The docs mention that you can add a table in the filter query:
> You can listen to individual rows using the format {table}:{col}=eq.{val} - where {col} is the column name, and {val} is the value which you want to match.

I've tried to use this feature but did not succeed:
```ts
    .on(
      "postgres_changes",
      {
        event: "UPDATE",
        schema: "public",
        table: "order_article",
        filter: "order:customer_id=eq." + customerId,
      },
```

I'm receiving the following error message in the realtime messages:
```
Subscribing to PostgreSQL failed: {:error, "Subscription insert failed with error: ERROR P0001 (raise_exception) invalid column for filter order:customer_id. Check that tables are part of publication supabase_realtime and subscription params are correct: %{\"event\" => \"UPDATE\", \"filter\" => \"order:customer_id=eq.8e0509fd-2555-4c8f-a088-e6f02be29552\", \"schema\" => \"public\", \"table\" => \"order_article\"}"}
```

Is this possible? And if so, how? Am I missing something obvious?

**Comment by GaryAustin1** [comment, marked as answer]
You can't use joins in realtime for either getting extra data or in filters.  
The only way around it is to use RLS to limit the rows visible to the user.

**Reply by jojomatik** [reply]
Could you explain what the `table` option in the docs means then? What can I achieve by setting it?
> You can listen to individual rows using the format {table}:{col}=eq.{val} - where {col} is the column name, and {val} is the value which you want to match.

**Reply by GaryAustin1** [reply]
Can you link where you see that?

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/realtime/postgres-changes?queryGroups=database-method&database-method=sql&queryGroups=language&language=js#filtering-for-specific-changes

**Reply by GaryAustin1** [reply]
Looks like that was the way v1 of realtime worked as then there was just the one string in .from() that included the table you want the realtime tracking on...
![image](https://github.com/supabase/supabase/assets/54564956/11506b77-d795-4f0c-bee6-3d8fc5920434)

**Reply by jojomatik** [reply]
Ah interesting, yes I got it from this page (the note is still there)

https://supabase.com/docs/reference/javascript/subscribe

![image](https://github.com/supabase/supabase/assets/66277921/264c049b-c13c-4214-908f-67d4c7e6090e)

So it's just an outdated note, that should've probably been removed?

**Reply by GaryAustin1** [reply]
Ahh,  Yeah that would be my guess is it just got copied.
